### PR TITLE
Limit maximum memory consumption of the logger

### LINKED ISSUE #794: Logger can consume unbound amount of memory
The logger lives in its own thread and receives events via a queue. This queue is currently unbound and can quickly fill up in a busy application. Especially when using log level trace. As a solution, we can give the queue a fixed size and block logging threads when full. This slows down the application, but at least it won't eventually crash when running out of memory.

### PR DESCRIPTION
**Pull request by Neverlord**
Give the logger queue a fixed size and block logging threads when full. This slows down the application, but at least it won't eventually crash when running out of memory.

Close #794.

I'm sorry for the large third commit. I should have done a better job at separating orthogonal changes. It's not that much code, though. So I hope it's still coherent enough for reviewing. :)

### CODE REVIEW

File: libcaf_core/caf/logger.hpp [hunk 1]
```diff
@@ -112,14 +112,17 @@ class logger : public ref_counted {
...
-          string_view fn, int line, std::string msg, std::thread::id t,
+    event(unsigned lvl, unsigned line, string_view cat, string_view full_fun,
+          string_view fun, string_view fn, std::string msg, std::thread::id t,
           actor_id a, timestamp ts);
 
     // -- member variables -----------------------------------------------------
 
     /// Level/priority of the event.
-    int level;
+    unsigned level;
```
Comment (josephnoir) [inline libcaf_core/caf/logger.hpp]: The function `accepts` still expects the level as an int.

File: configure [hunk 2]
```diff
@@ -283,27 +283,7 @@ while [ $# -ne 0 ]; do
...
-                    ;;
-                TRACE)
-                    append_cache_entry CAF_LOG_LEVEL STRING 4
-                    ;;
-                *)
-                    echo "Invalid log level '$level'. Try '$0 --help' to see valid values."
-                    exit 1
-                    ;;
-            esac
+            append_cache_entry CAF_LOG_LEVEL STRING "$level"
```
[the remark this answers is not in the document]
Comment (Neverlord) [reply configure]: You won't get an error from CMake, but of course the compiler will complain pretty much immediately.

File: libcaf_core/caf/logger.hpp [hunk 3]
```diff
@@ -66,16 +65,59 @@ namespace caf {
...
 class logger : public ref_counted {
 public:
+  // -- friends ----------------------------------------------------------------
+
   friend class actor_system;
 
+  // -- constants --------------------------------------------------------------
+
+  /// Configures the size of the circular event queue.
+  static constexpr size_t queue_size = 128;
```
[the remark this answers is not in the document]
Comment (Neverlord) [reply libcaf_core/caf/logger.hpp]: Slowing down the application when it runs too fast for the logger to keep up is the point of this PR. We only want to be able to absorb small log event bursts. Actually, I considered cutting the queue size back to 64. The only reason I didn't is that this is a low per-core event buffer when running on high-end machines with dozens of cores.

File: libcaf_core/caf/logger.hpp (hunk 1 above)
Comment (Neverlord) [reply libcaf_core/caf/logger.hpp]: Good catch. Fixed.

**Pull request merged**